src/trackedit/internal/trackeditactionscontroller.cpp

```diff
@@ -540,6 +545,26 @@ void TrackeditActionsController::newLabelTrack()
     trackeditInteraction()->newLabelTrack();
 }
 
+void TrackeditActionsController::deleteTrack(const muse::actions::ActionData& args)
+{
+    TrackId trackId { -1 };
+    if (args.count()) {
+        // Call from the context menu
+        trackId = args.arg<TrackId>(0);
+    } else {
+        // Call from shortcut
+        trackId = selectionController()->selectedTrack();
+    }
+
+    if (trackId == -1) {
+        return;
+    }
+
+    trackeditInteraction()->deleteTrack(trackId);
+
+    projectHistory()->pushHistoryState("Delete track", "Delete track");
```

Comment:
This should be called in trackeditInteraction()->deleteTrack.
The point is that if trackeditInteraction()->deleteTrack is called from somewhere else.. the logic associated with the history should also be executed, i.e. it is part of the logic of the trackeditInteraction()->deleteTrack method

Reply:
Actually I started to think about that in terms of implementing clip overlapping: when you move a clip over another clip in the end you end up with doing multiple actions at once:
 1. Clip move
2. Clip trim or clip delete
Both of these will add state to the undo-stack while I think it should be considered as one action
What do you think?

Reply:
There is also this option: 

```
void moveClip(.., bool isModifyHistory = true)
{
    //  do move clip
    ...
    
    if (isModifyHistory) {
        pushHistoryState(...);
    }
}

void overlapClip(.., bool isModifyHistory = true)
{
   moveClip(.., false /*isModifyHistory*/);
   trimClip(.., false /*isModifyHistory*/));
   
   if (isModifyHistory) {
       pushHistoryState(...);
   }
}

```
but the first option with private methods without side effects is better